packages/warning/src/index.js

```diff
@@ -0,0 +1,45 @@
+function isDev() {
+	return (
+		typeof process !== 'undefined' &&
+		process.env &&
+		process.env.NODE_ENV !== 'production'
+	);
+}
+
+/**
+ * Shows a warning with `messages` if `condition` passes and environment is not `production`.
+ *
+ * @param {boolean} condition Whether the warning will be triggered or not.
+ * @param {string[]} messages Message(s) to show in the warning.
+ *
+ * @example
+ * ```js
+ * import warning from '@wordpress/warning';
+ *
+ * function MyComponent( props ) {
+ *   warning( ! props.title, '`props.title` was not passed' );
+ *   ...
+ * }
+ * ```
+ */
+export default function warning( condition, ...messages ) {
+	if ( isDev() ) {
```

Comment:
Minor: You could avoid indentation on the entire function with an early return (similar to what's done on the immediate next line with `if ( ! condition ) {`):

```js
if ( ! isDev() ) {
	return;
}
```

Alternatively, you could assign the function itself as a noop for the static export member, rather than evaluating each time it's called (not that it's a very expensive evaluation):

```js
const warning = isDev() ?
	() => {} :
	( condition, ...messages ) => {
		// ...
	};

export default warning;
```